cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/operationrunner/PullAppContext.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.sourcecontrol.operationrunner;
+
+import io.cdap.cdap.proto.id.ApplicationReference;
+import io.cdap.cdap.proto.sourcecontrol.RepositoryConfig;
+
+/**
+ * Information required by {@link SourceControlOperationRunner} to execute the task of
+ * pulling an application from linked repository.
+ */
+public class PullAppContext {
```

Comment:
nit: let's name it PullAppRequest instead of Context. Context in the project has referred to classes in the api/spi modules rather than internal classes. It's usually used because it contains a mix of a bunch of different platform APIs so there isn't a great name. Internal classes like this can have a more descriptive and focused name.

Reply:
Makes sense. I was following existing conventions. Will update for all SCM request objects